modules/gapi/samples/one_vpl_source_to_bgr_conv.cpp

```diff
@@ -0,0 +1,224 @@
+#endif // HAVE_DIRECTX
+}
+
+const std::string about =
+    "This is an example that covers case when a MediaFrame is received from GPU using VPL Source and converted to BGR UMat";
+const std::string keys =
+    "{ h help                       |                                           | Print this help message }"
+    "{ input                        |                                           | Path to the input video file }"
+    "{ output                       |                                           | Path to the output RAW video file. Use .avi extension }";
+
+int main(int argc, char *argv[]) {
+    cv::CommandLineParser cmd(argc, argv, keys);
+    cmd.about(about);
+    if (cmd.has("help")) {
+        cmd.printMessage();
+        return 0;
+    }
+#ifdef HAVE_DIRECTX
+#ifdef HAVE_D3D11
+    // Get file name
+    const auto file_path = cmd.get<std::string>("input");
+    const auto output = cmd.get<std::string>("output");
+
+    // check ouput file extension
+    if (!output.empty()) {
+        auto ext = output.find_last_of(".");
+        if (ext == std::string::npos || (output.substr(ext + 1) != "avi")) {
+            std::cerr << "Output file should have *.avi extension for output video" << std::endl;
+            return -1;
+        }
+    }
+
+    auto adapter_factory = createCOMPtrGuard<IDXGIFactory>();
+    {
+        IDXGIFactory* out_factory = nullptr;
+        HRESULT err = CreateDXGIFactory(__uuidof(IDXGIFactory),
+                                    reinterpret_cast<void**>(&out_factory));
+        if (FAILED(err)) {
+            std::cerr << "Cannot create CreateDXGIFactory, error: " << HRESULT_CODE(err) << std::endl;
+            return -1;
+        }
+        adapter_factory = createCOMPtrGuard(out_factory);
+    }
+
+    auto intel_adapter = createCOMPtrGuard<IDXGIAdapter>();
+    UINT adapter_index = 0;
+    const unsigned int refIntelVendorID = 0x8086;
+    IDXGIAdapter* out_adapter = nullptr;
+
+    while (adapter_factory->EnumAdapters(adapter_index, &out_adapter) != DXGI_ERROR_NOT_FOUND) {
+        DXGI_ADAPTER_DESC desc{};
+        out_adapter->GetDesc(&desc);
+        if (desc.VendorId == refIntelVendorID) {
+            intel_adapter = createCOMPtrGuard(out_adapter);
+            break;
+        }
+        ++adapter_index;
+    }
+
+    if (!intel_adapter) {
+        std::cerr << "No Intel GPU adapter on aboard. Exit" << std::endl;
+        return -1;
+    }
+    auto dx11_dev = createCOMPtrGuard<ID3D11Device>();
+    auto dx11_ctx = createCOMPtrGuard<ID3D11DeviceContext>();
+    std::tie(dx11_dev, dx11_ctx) = create_device_with_ctx(intel_adapter.get());
+    void* accel_device_ptr = nullptr;
+    void* accel_ctx_ptr = nullptr;
+    accel_device_ptr = reinterpret_cast<void*>(dx11_dev.get());
+    accel_ctx_ptr = reinterpret_cast<void*>(dx11_ctx.get());
+
+    cv::Ptr<cv::gapi::wip::IStreamSource> cap;
+    std::vector<cv::gapi::wip::onevpl::CfgParam> source_cfgs;
+    source_cfgs.push_back(cfg::create_from_string(
```

Comment:
you can drop `create_from_string` at all and reuse `source_cfgs.push_back(CfgParam::create_acceleration_mode("MFX_ACCEL_MODE_VIA_D3D11"));`